Readme.md

```diff
@@ -1,21 +1,69 @@
-# Lombiq <add project name here>
+# Lombiq PowerShell Analyzers
 
 
 
 ## About
 
-Add a general overview of the project here. Keep or remove the OSOCE note below as necessary.
+Powershell static code analysis via [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) and [our configuration for it](PSScriptAnalyzerSettings.psd1).
 
 Do you want to quickly try out this project and see it in action? Check it out in our [Open-Source Orchard Core Extensions](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions) full Orchard Core solution and also see our other useful Orchard Core-related open-source projects!
 
 
 ## Documentation
 
-Add detailed documentation here. If it's a lot of content then create documentation pages under the *Docs* folder and link pages here.
+### Pre-requisites
 
+The *PSScriptAnalyzer* module must be installed. If it's not installed, then running the script in `pwsh` will output no results and running it in `powershell` will throw an exception. Follow the steps [here](https://docs.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/overview?view=ps-modules#installing-psscriptanalyzer).
+
+Note that if you are usnig this in GitHub Actions, the common images (`windows-latest` and `ubuntu-latest`) already have it so you don't need to install anything.
+
+### Usage
+
+#### PowerShell CLI
+
+Use the script like this to output the analyzer violations with `Write-Error`:
+
+```pwsh
+./Invoke-Analyzer.ps1 -SettingsPath PSScriptAnalyzerSettings.psd1
+```
+
+The `-SettingsPath` can be omitted, in this case the _PSScriptAnalyzerSettings.psd1_ in the same directory as the _Invoke-Analyzer.ps1_ will be used.
+
+#### GitHub Actions
+
+You can invoke it from an _action.yml_ file like this:
+```yaml
+    - name: Lint PowerShell scripts
+      shell: pwsh
+      run: ${{ github.action_path }}/Invoke-Analyzer.ps1 -ForGitHubAction
+```
+
+The `-ForGitHubAction` optional displays the results using [error workflow commands](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message). These create file annotations pointing to the exact script path and line number provided by PSScriptAnalyzer. You can review the results in the workflow summary page. If the violating files aren't in a submodule then they will be marked in the related pull request's Files tab as well.
+
+If you are using our `build-dotnet` action or building-related reusable workflows from [Lombiq GitHub Actions](https://github.com/Lombiq/GitHub-Actions), PowerShell linting is already included.
+
+Just set the value of the `powershell-analyzer-path` to the path of the _Invoke-Analyzer.ps1_ file relative to your repository root. In case it's _./tools/Lombiq.Analyzers.PowerShell/Invoke-Analyzer.ps1_ you don't need additional configuration.
+
+#### MSBuild
```

Comment:
Explain how you'd add it to a solution like OSOCE.

Reply:
Already done [here](https://github.com/Lombiq/PowerShell-Analyzers/blob/4c113a69baa3c5b3fcc5530c0425632ffecabd0b/Readme.md?plain=1#L51-L55). What are you missing?